### PR TITLE
Primitive crafting overhaul

### PR DESCRIPTION
**Pull request by Rail-Runner**
#### Summary

SUMMARY: Content "Expand primitive crafting"
#### Purpose of change

Previously, recipes involving rocks didn't account for there being multiple kinds of rocks (at least as far as stoneworking is concerned); some can flake if stuck, others can't. This PR fixes that, and fixes a few other weird things about primitive tools.
#### Describe the solution

Add primitive stoneworking tools: billets, punches and rock drill.
Migrate stone chisel item from Magiclysm, for primitive woodworking.
Change stone shovel to  wooden shovel. I couldn't find any info about stone shovels anywhere (and they don't seem any more practical than a digging stick to me either), so I changed it to a wooden shovel, which as I could figure out was actually a thing in the past.
Alter existing primitive tool recipes, so as to reflect the general method of creating them; either ground stone tools, or flaked stone tools.
Change stone quern to be much bulker and heavier. Presumably it's made from large solid rocks; it can't be this light, if so.
Primitive stoneworking recipes in general were changed to require much more skill; as I could figure out, it isn't really something you can figure out quickly.
Made some minor alterations to existing tools, reflecting the limits they'd have in real life, and specifying length for them.

#### Describe alternatives you've considered

Removing stone shovel completely; though seems like wooden shovel could take its place now.
#### Testing

Loaded the game, made sure that no errors happen.
#### Additional context

None.

### CODE REVIEW

**Comment by I-am-Erk** [comment]
I am interested in this, but one of the issues we've constantly faced with stone tools is that knapping is not at all a simple skill and can take a very long time to learn. Previously I've recommended adding a specific knapping proficiency (requiring therefore a proficiency system) to do it. I would be open to at least temporarily doing the same thing we have with helicopter pilots, and modeling the same thing with a mutation (neolithic revivalist?) That allows you to do knapping recipes, or maybe reduces the time dramatically - in which case without the trait, they should be very very very slow. Then, at least, we could have ztarting professions that get these things as a specific boon

**Comment by kevingranade** [comment]
If you place knapping stones in all the same places and frequencies where non-knapping stones appear, there's no reason to have the distinction.

At a minimum knapping stones need to be significantly rarer than non-knapping stones, and ideally they would also be limited to locations where players would need to seek them out.

**Comment by EvgenijM86** [comment]
When using Staff Sling or Sling I noticed that when collecting rocks I collect a lot sharp rocks that I have no use for. Maybe add them as ammo for Staff Sling/Sling for less bash damage and more cut damage?

**Comment by dissociativity** [comment]
I'm not sure that sharp stones would be particularly aerodynamic at all, but they certainly need more use cases, perhaps knapping being more of a thing will make already sharp stones more useful to players that don't want to go down that rabbit hole?

**Comment by Rail-Runner** [comment]
> At a minimum knapping stones need to be significantly rarer than non-knapping stones, and ideally they would also be limited to locations where players would need to seek them out.

Done, I think.

**Comment by kevingranade** [comment]
Causes a failure in the melee DPS test.

**Comment by Rail-Runner** [comment]
Seems like it's fixed now.

File: data/json/items/tool/knives.json [hunk 1]
```diff
@@ -92,5 +92,21 @@
     "color": "light_gray",
     "qualities": [ [ "CUT", 1 ], [ "BUTCHER", 11 ] ],
     "flags": [ "STAB", "SHEATH_KNIFE" ]
+  },
+  {
+    "type": "GENERIC",
```
Comment (Wishbringer) [inline data/json/items/tool/knives.json:97]:
```suggestion
    "type": "GENERIC",
```
"type": "TOOL",

This item should probably be a tool?

**Pull request merged**